### ISSUE TITLE
test(chacha): NIVC full test with 40 word fold

### ISSUE THREAD

**Issue by Autoparallel** [pluto/web-prover-circuits#59]
## Tasks
- [ ] Create a an additional test for `NIVC_FULL_2_CHACHA` that uses a param of `[40]` (the other test uses `[80]`

This assures that we can fold multiple portions of plaintext/ciphertext together in a coherent way. We need to make sure this is doable so that, in the future, we can handle plaintext sizes >1KB.

Potentially: 
- [ ] Test that this also handles padding appropriately (this would be a separate issue).

**Comment by 0xJepsen** [comment]
I don't think it makes sense to fold chacha at all. It already is so much faster and will not be a bottle neck.

**Comment by 0xJepsen** [comment]
In googles research they found average response sizes we between 500b to 2kb. 
https://www.chromium.org/spdy/spdy-whitepaper/

**Comment by 0xJepsen** [comment]
what is an example of a response that would be 10k, i feel like less 0.01% responses are that big.

**Comment by 0xJepsen** [comment]
> what is an example of a response that would be 10k, i feel like less 0.01% responses are that big.

**Comment by devloper** [comment]
I don't see this as a launch blocker for us today, but I do think it'll be common to have API requests in the range of 1-10KB.  In my mind where this becomes _most_ impactful is when we start attempting to prove HTML content.  Those pages are much bulkier and range be from hundreds of kilobytes to megabytes!! 

For example, try:

`curl -so /dev/null https://www.wsj.com -w '%{size_download}'`

This is 500kb for the homepage of wsj

imo, we don't need to have this for launch but as a vision and goal larger proofs is definitely a priority to me.

**Comment by 0xJepsen** [comment]
Nice yeah that makes sense to me.

**Comment by Autoparallel** [comment]
Part of the circuit roadmap is to allow >1KB proofs. Therefore, we will need to fold ChaCha. Hence, we need this done!

Not immediately, but roughly by v0.8. Go look at the circuits roadmap to see the plan moving forward

**Comment by Autoparallel** [comment]
E.g., prove authorship of a paper on arXiv:
```
❯ curl -H "Accept: application/xml" \
"https://export.arxiv.org/api/query?id_list=2401.03703" \
--silent --output /dev/null --write-out "Size: %{size_download} bytes\n"

Size: 3260 bytes
```

**Comment by 0xJepsen** [comment]
Chacha with 1kb data is already faster than folded aes for 1kb so i don't think it's needed for 1kb.

**Comment by Autoparallel** [comment]
> Chacha with 1kb data is already faster than folded aes for 1kb so i don't think it's needed for 1kb.

Correct, but that's why both of us have said it isn't a blocker today

**Comment by Autoparallel** [comment]
i'm now getting rekt by this
